lib/yang_wrappers.c

```diff
@@ -676,6 +677,60 @@ void yang_get_default_string_buf(char *buf, size_t size, const char *xpath_fmt,
 			  xpath);
 }
 
+/*
+ * Primitive type: binary.
+ */
+struct yang_data *yang_data_new_binary(const char *xpath, const char *value,
+				       size_t len)
+{
+	char *value_str;
+	struct base64_encodestate s;
+	int cnt;
+	char *c;
+
+	value_str = (char *)malloc(len * 2);
+	base64_init_encodestate(&s);
+	cnt = base64_encode_block(value, len, value_str, &s);
+	c = value_str + cnt;
+	cnt = base64_encode_blockend(c, &s);
+	c += cnt;
+	*c = 0;
+
+	struct yang_data *data = yang_data_new(xpath, value_str);
+	free(value_str);
+	return data;
+}
+
+void yang_dnode_get_binary_buf(char *buf, size_t size,
```

Comment:
This function is unusable right now. The `buf` is populated with some binary data (not zero-terminated), and the caller has no idea about the length of this data. I propose to use the actual length as a return value.

Reply:
updated